File: app/src/main/java/com/imangazalievm/bubbble/di/UserFollowersPresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.UserFollowersPresenterModule;
import com.imangazalievm.bubbble.di.modules.UserProfilePresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.UserFollowersPresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = UserFollowersPresenterModule.class)
public interface UserFollowersPresenterComponent {

    UserFollowersPresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/userprofile/UserFollowersPresenterComponent.kt
package com.imangazalievm.bubbble.di.userprofile

import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.presentation.userprofile.followers.UserFollowersPresenter
import dagger.Component

@Presenter
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [UserFollowersPresenterModule::class]
)
interface UserFollowersPresenterComponent {

    fun getPresenter(): UserFollowersPresenter

}